pkg/logs/input/docker/launcher.go

```diff
@@ -159,13 +173,68 @@ func (l *Launcher) startTailer(container *Container, source *config.LogSource) {
 	}
 
 	// keep the tailer in track to stop it later on
-	l.tailers[containerID] = tailer
+	l.addTailer(containerID, tailer)
 }
 
 // stopTailer stops the tailer matching the containerID.
 func (l *Launcher) stopTailer(containerID string) {
 	if tailer, isTailed := l.tailers[containerID]; isTailed {
 		go tailer.Stop()
-		delete(l.tailers, containerID)
+		l.removeTailer(containerID)
+	}
+}
+
+func (l *Launcher) restartTailer(containerID string) {
+	log.Warnf("Tailing will restart for %v", ShortContainerID(containerID))
```

Comment:
Not necessarily, depending on how it goes.
I think we should either include the notion that we are trying, or move/remove this log.